console/v1/types_console_plugin.go

```diff
@@ -0,0 +1,197 @@
+)
+
+// ConsolePluginI18n holds information on localization resources that are served by
+// the dynamic plugin.
+type ConsolePluginI18n struct {
+	// load indicates how the plugin's localization resource should be loaded.
+	// +kubebuilder:validation:Required
+	Load LoadType `json:"load"`
+}
+
+// ConsolePluginProxy holds information on various service types
+// to which console's backend will proxy the plugin's requests.
+type ConsolePluginProxy struct {
+	// backend provides information about endpoint to which the request is proxied to.
+	Endpoint ConsolePluginProxyEndpoint `json:"endpoint"`
+	// alias is a proxy name that identifies the plugin's proxy. An alias name
+	// should be unique per plugin. The console backend exposes following
+	// proxy endpoint:
+	//
+	// /api/proxy/plugin/<plugin-name>/<proxy-alias>/<request-path>?<optional-query-parameters>
+	//
+	// Request example path:
+	//
+	// /api/proxy/plugin/acm/search/pods?namespace=openshift-apiserver
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	// +kubebuilder:validation:Pattern=`^[A-Za-z0-9-_]+$`
+	Alias string `json:"alias"`
+	// caCertificate provides the cert authority certificate contents,
+	// in case the proxied Service is using custom service CA.
+	// By default, the service CA bundle provided by the service-ca operator is used.
+	// +kubebuilder:validation:Pattern=`^-----BEGIN CERTIFICATE-----([\s\S]*)-----END CERTIFICATE-----\s?$`
+	// +kubebuilder:validation:Optional
+	CACertificate string `json:"caCertificate,omitempty"`
+	// authorization provides information about authorization type,
+	// which the proxied request should contain
+	// +kubebuilder:validation:Optional
+	Authorization AuthorizationType `json:"authorization,omitempty"`
+}
+
+// ConsolePluginProxyEndpoint holds information about the endpoint to which
+// request will be proxied to.
+type ConsolePluginProxyEndpoint struct {
+	// type is the type of the console plugin's proxy. Currently only "Service" is supported.
+	// +kubebuilder:validation:Required
+	Type ConsolePluginProxyType `json:"type"`
+	// service is an in-cluster Service that the plugin will connect to.
+	// The Service must use HTTPS. The console backend exposes an endpoint
+	// in order to proxy communication between the plugin and the Service.
+	// Note: service field is required for now, since currently only "Service"
+	// type is supported.
+	// +kubebuilder:validation:Required
+	Service ConsolePluginProxyServiceConfig `json:"service"`
+}
+
+// ProxyType is an enumeration of available proxy types
+// +kubebuilder:validation:Pattern=`^(Service)$`
+type ConsolePluginProxyType string
+
+const (
+	// ProxyTypeService is used when proxying communication to a Service
+	ProxyTypeService ConsolePluginProxyType = "Service"
+)
+
+// AuthorizationType is an enumerate of available authorization types
+// +kubebuilder:validation:Pattern=`^(UserToken|None)$`
```

Comment:
Use an enum style tag
```suggestion
// +kubebuilder:validation:Enum:=UserToken;None
```